ote/src/cljs/ote/app/controller/front_page.cljs

```diff
@@ -21,17 +22,34 @@
 (defrecord TransportOperatorDataResponse [response])
 (defrecord TransportOperatorDataFailed [error])
 
+(defn navigate [event {:keys [before-unload-message navigation-prompt-open?] :as app} navigate-fn]
+  (if (and before-unload-message (not navigation-prompt-open?))
+    (assoc app
+           :navigation-prompt-open? true
+           :navigation-confirm event)
+    (navigate-fn (dissoc app
+                         :navigation-prompt-open?
+                         :before-unload-message
+                         :navigation-confirm))))
+
 (extend-protocol tuck/Event
 
   ChangePage
-  (process-event [{given-page :given-page} app]
-    (routes/navigate! given-page)
-    (assoc app :page given-page))
+  (process-event [{given-page :given-page :as e} app]
+    (navigate e app (fn [app]
+                      (do
+                        (routes/navigate! given-page)
+                        (assoc app :page given-page)))))
 
   GoToUrl
-  (process-event [{url :url} app]
-    (set! (.-location js/window) url )
-    app)
+  (process-event [{url :url :as e} app]
+    (navigate e app (fn [app]
+                      (set! (.-location js/window) url)
```

Comment:
What happens our app-state when url is changed?
It doesn't refresh it from the server?

Reply:
On hash change, nothing that we don't specifically do in routes/on-navigate.